### PR TITLE
Add version to the InstrumenterBuilder and Instrumenter

### LINKED ISSUE #4599: Hardcoded Tracer version in Instrumenter
**Is your feature request related to a problem? Please describe.**
In https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/d314d76fce9400976d1dfb8c326ab3d9394991d4/instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java#L84-L85

The `Instrumenter` constructor has a hardcoded version to retrieve the `Tracer`. While this works fine for instrumentation libraries inside the project, it forces external libraries to use the same hardcoded version if you want to retrieve the same `Tracer` used by your `Instrumenter`.

**Describe the solution you'd like**
Have the `Tracer` version be part of the `InstrumenterBuilder` so it can be set by a consumer. If not set in the builder it defaults to the current behavior (use the OTel version).

**Describe alternatives you've considered**
None.

**Additional context**
I'm happy to provide a PR if this is an acceptable change.

### PR DESCRIPTION
**Pull request by radcortez**
- Fixes #4599

### CODE REVIEW

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java [hunk 1]
```diff
@@ -58,7 +58,17 @@
       OpenTelemetry openTelemetry,
       String instrumentationName,
       SpanNameExtractor<? super REQUEST> spanNameExtractor) {
-    return new InstrumenterBuilder<>(openTelemetry, instrumentationName, spanNameExtractor);
+    return new InstrumenterBuilder<>(
+        openTelemetry, InstrumentationVersion.VERSION, instrumentationName, spanNameExtractor);
```
Comment (laurit) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java:62]: I think you have switched name and version here

Comment (iNikem) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java:62]: @radcortez Please add new tests as well

**Review by iNikem — CHANGES_REQUESTED** [review]

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java (hunk 1 above)
Comment (radcortez) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java]: Ops.

**Review by iNikem — APPROVED** [review]

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java [hunk 2]
```diff
@@ -58,7 +58,17 @@
       OpenTelemetry openTelemetry,
       String instrumentationName,
       SpanNameExtractor<? super REQUEST> spanNameExtractor) {
-    return new InstrumenterBuilder<>(openTelemetry, instrumentationName, spanNameExtractor);
+    return new InstrumenterBuilder<>(
+        openTelemetry, instrumentationName, InstrumentationVersion.VERSION, spanNameExtractor);
+  }
+
+  public static <REQUEST, RESPONSE> InstrumenterBuilder<REQUEST, RESPONSE> builder(
```
Comment (mateuszrzeszutek) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java:65]: Can you add some Javadoc to this method? Just copy everything from the 3-arg `builder()` and mention that `instrumentationVersion` refers to the instrumentation library version, not instrumented library version.

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java (hunk 2 above)
Comment (radcortez) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java:65]: Sure.

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java [hunk 3]
```diff
@@ -58,7 +58,35 @@
...
+   * <p>The {@code instrumentationVersion} is the version of the instrumentation library, not the
+   * version of the instrument*ed* library.
+   *
+   * <p>In this project we use a convention to encode the minimum supported version of the
+   * instrument*ed* library into the instrumentation name, for example {@code
+   * io.opentelemetry.apache-httpclient-4.0}. This way, if there are different instrumentations for
+   * different library versions it's easy to find out which instrumentations produced the telemetry
+   * data.
+   */
+
```
Comment (laurit) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java:82]:
```suggestion

```
We don't usually have a blank line between javadoc and method declaration

**Review by laurit — APPROVED** [review]

**Comment by trask** [comment]
Thanks @radcortez!

**Pull request merged**